core/src/main/kotlin/net/corda/core/contracts/Structures.kt

```diff
@@ -340,3 +344,74 @@ class PrivacySalt(bytes: ByteArray) : OpaqueBytes(bytes) {
  */
 @KeepForDJVM
 data class StateAndContract(val state: ContractState, val contract: ContractClassName)
+
+/**
+ * A [StatePointer] contains a [pointer] to a [ContractState]. The [StatePointer] can be included in a [ContractState]
+ * or included in an off-ledger data structure. [StatePointer]s can be resolved to a [StateAndRef] by performing a
+ * vault query. There are two types of pointers; linear and static. [LinearPointer]s are for use with [LinearState]s.
+ * [StaticPointer]s are for use with any type of [ContractState].
+ */
+@CordaSerializable
+interface StatePointer {
+    /**
+     * An identifier for the [ContractState] that this [StatePointer] points to.
+     */
+    val pointer: Any
+
+    /**
+     * Resolves a [StatePointer] to a [StateAndRef] via a vault query. This method will either return a [StateAndRef]
+     * or return an exception.
+     *
+     * @param services a [ServiceHub] implementation is required to resolve the pointer.
+     */
+    fun resolve(services: ServiceHub): StateAndRef<ContractState>
+}
+
+/**
+ * A [StaticPointer] contains a [pointer] to a specific [StateRef] and can be resolved by looking up the [StateRef] via
+ * [ServiceHub]. There are a number of things to keep in mind when using [StaticPointer]s:
+ * - The [ContractState] being pointed to may be spent or unspent when the [pointer] is resolved
+ * - The [ContractState] may not the known by the node performing the look-up in which case the [resolve] method will
```

Comment:
"may not the known by the node" -> typo
```suggestion
 * - The [ContractState] may not be known by the node performing the look-up in which case the [resolve] method will
```